packages/flutter/test/gestures/gesture_binding_test.dart

```diff
@@ -230,21 +258,21 @@ void main() {
       expect(events[2], isA<PointerDownEvent>());
       expect(events[2].buttons, equals(kPrimaryButton));
       expect(events[3], isA<PointerMoveEvent>());
-      expect(events[3].buttons, equals(kPrimaryButton | kSecondaryStylusButton));
+      expect(events[3].buttons, equals(kSecondaryStylusButton));
```

Comment:
This is interesting. So I was aware that the primary button is synthesized besides the secondary. I think the intention was that: This case indicates that the stylus first touches, then the user presses the button, in which case `kPrimaryButton | kSecondaryStylusButton` should be the correct result. Do you think we should somehow handle this case, or does it conflict with what you recorded on real device?

Reply:
I think most styluses that have a right-click button such as Surface Pen don't let you change the button during a press. I have only been able to get it to work by holding the button before touching the screen. So maybe I should split this test case as well?

Changing the buttons during a pointer's contact should be allowed, but we should rely on the engine to send an event with the correct buttons set. The point of this synthesis is to ensure we never have buttons = 0, if we always add the primary button, right-clicking on widgets doesn't work at all with a stylus.

Reply:
Ok, makes sense.

Reply:
So should I split up this case too? I have a feeling these cases were not originally meant to be real continuous sequences, for example, hover events do not make much sense for touch devices.

Reply:
No I was imagining. Let's keep this case until someone complains.